test/derivatives/ArrayFunctionTests.jl

```diff
@@ -27,25 +27,42 @@ function testcat(f, args::Tuple, type, kwargs=NamedTuple())
         @test value(x) == f(args...; kwargs...)
     else
         @assert length(args) == 2
-        x = f(track(args[1]), args[2]; kwargs...)
-        @test x isa type
-        @test value(x) == f(args...; kwargs...)
 
-        x = f(args[1], track(args[2]); kwargs...)
-        @test x isa type
-        @test value(x) == f(args...; kwargs...)
+        broken = f == hcat && (args[2] isa AbstractMatrix)
+        if broken && VERSION >= v"1.4"
```

Comment:
Why 1.4? AFAICT (and as mentioned eg in #259) this issue is caused by a type piracy in SparseArrays introduced in Julia 1.10 (see https://github.com/JuliaSparse/SparseArrays.jl/issues/431).

Reply:
I didn't do the bisection to figure out which version of Julia was guilty. I just took the version that was above 1.3, on which I was sure the tests were passing.
Tbh we should just drop 1.0 and 1.3, even Julia doesn't support those anymore